### DISCUSSION TITLE
Status Updates (Week 5)

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#45] [General]
Respond to this thread with everything you've worked on from 4/15/21-4/22/21.

What did you work on this week?

What sort of technical blockers are you facing?

What are you going to work on next week?

What sort of new technologies did you learn, if any?

Put any links to commits/other contributions here.

Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?

Feel free to add more questions as you think of them!

**Comment by roryeiffe** [comment]
## What did you work on this week?

This week, I was kind of busy with other classes, but I was able to start working on the custom word integration. Halfway through however, I realized I needed to merge appState with main before I could make any real progress. So, Josh and Abdul helped me resolve merge conflicts. Afterwards, I made sure everything was working on the AppState branch and merged it back into main. Now that everything I need is on the main branch, I will be able to branch off and implement the custom word integration. Finally, I updated the wiki to contain some general information about the project and features that we have added. 

## What sort of technical blockers are you facing?

The main blocker of this week was figuring out how to resolve the merge conflicts. I had to wait for our team meeting to be able to discuss those. 

## What are you going to work on next week?

Now that main is up to date with appState, I will be able to work on custom word integration. I think I can get it done by the time we have to present on Tuesday. 

## What sort of new technologies did you learn, if any?

No new technologies this week. 

## Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/issues/37
https://github.com/roryeiffe/Spoiler-Block/commit/2840ce02c2f4f9e989dec56a89420a90024a5c29
https://github.com/roryeiffe/Spoiler-Block/commit/43c0453b2fda7eaabacd2e0a132f797c568a22dc
https://github.com/roryeiffe/Spoiler-Block/wiki/Code-Overview
https://github.com/roryeiffe/Spoiler-Block/wiki
https://github.com/roryeiffe/Spoiler-Block/pull/41/
https://github.com/roryeiffe/Spoiler-Block/issues/48

## Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?

I am proud that we have a working extension and something cool to show off during our presentation, especially since this was a brand new project. I am also really proud of everyone's contributions and our ability to work well as a team.

**Comment by SerenaChen1** [comment]
## What did you work on this week?

This week, Abdul and I created a mockup for the popup that will warn the user of the possibility of spoilers on a given webpage. We came up with a single design, but we each chose a different color scheme for it, so our team will vote between the two the following Monday (4/26). Then, I worked on the message passing for the threshold values. This will come in handy once that popup is made so that we can use that value to warn the user about spoilers if the possibility of spoilers is greater than the saved threshold percentage.

## What sort of technical blockers are you facing?

I didn't really face any technical blockers this week because I was mainly working on Figma mockups and implementing a feature that we already had a similar version of.

## What are you going to work on next week?

I actually missed the UI team meeting and general meeting on Thursday (4/22) because I had my COVID vaccine appointment, so I wasn't assigned any new tasks, but I think I'd like to help out on getting the warning popup created and functioning. 

## What sort of new technologies did you learn, if any?

I didn't learn any new technologies this week.. 

## Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/commit/0a522ad6686be58bc91bbfb188ef4c7fdfedc1fe
https://github.com/roryeiffe/Spoiler-Block/issues/50
https://www.figma.com/file/N6RD9WowT8LbuWFit0kE9f/Spoiler-Block?node-id=0%3A1

## Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?

I am proud our team put together a solid Chrome extension in the time we were given. I think, especially for the UI team, we did a great job meeting multiple times a week and dividing tasks. Each meeting went smoothly in that we each shared what we worked on and then discussed what we plan on doing for the following week. Before this, I had very little experience with React, so I'm glad I was able to brush up on those skills and learn more about message passing, which I found pretty interesting.

**Comment by VictoriaSavage526** [comment]
### What did you work on this week?
I worked on modifying colors to be in line with our chosen color themes in CSS. I also started working on integrating a logo for the extension pop up. I also merged some Pull Requests

### What sort of technical blockers are you facing?
None at the moment.

### What are you going to work on next week?
I will work with Abdul and Serena on the warning popup, and continue working to integrate the logo into the popup and making everything in theme.

### What sort of new technologies did you learn, if any?
I learned more about CSS

### Put any links to commits/other contributions here.
*TBD*

### Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?
I'm really proud about how we all worked together. This group has been great to work with, and we created a serviceable extension in a short amount of time

**Comment by BreadGarLick** [comment]
### What did you work on this week?

I wrote code to detect if google autocomplete elements are present on google.com, and if there is, to re move it.
This prevents a spoiler form appearing while google searching.

### What sort of technical blockers are you facing?

I was having issues with getting the code to identify the correct page elements, but I've fixed that.

### What are you going to work on next week?

Refining the blocking to be more elegant and to add more features. I'll also help out with any open issues.

### What sort of new technologies did you learn, if any?

Leaning more about page elements and IDs was interesting, as I blocked the google autocomplete elements by their IDs.

### Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/pull/47

### Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?
I'm super impressed with how much we got done in these few weeks. We've created a working extension with a great UI. It's been really cool to see it all come together.

**Comment by TheCovenant** [comment]
### What did you work on this week:

This week I met with the backend team (Josh, Ben) along with Rory to inspect the backend and discuss how to integrate it with the frontend. I learned a lot about the patterns being used in the backend. I was unable to put a lot of time into the actual integration.

### What sort of technical blockers are you facing:

I'm not facing any significant technical blockers at this point. 

### What are you going to work on next week:

One thing seriously lacking in our codebase is comments. I plan to add some clarifying comments that should prove useful in explaining what is seen in the codebase.

### What sort of new technologies did you learn, if any:
None

## Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?

I am really proud of how all members of the group were able to coordinate with each other and find their role in making this project a reality. Discussions were courteous and productive, and we were able to have a clear direction of the way forward almost all the time.

### Links to Contributions:

**Comment by Sushiman33** [comment]
### What did you work on this week?

This week I made the fix for blocking words with a dictionary lookup instead of looping through and array and added more words to the
 
### What sort of technical blockers are you facing?
Only problems that are still persisting is if there is a feature of javascript that I still don't understand as well. 

### What are you going to work on next week?

Fix some issues that can be fixed in content.js

### Put any links to commits/other contributions here.
https://github.com/roryeiffe/Spoiler-Block/pull/38/commits/c0ff06b77d4631f87c1645087e5891b2c3ca1c5a
https://github.com/roryeiffe/Spoiler-Block/pull/43/commits/20d6abbc134fef52407441a53a6fd2dd9a540f62

### Since we're getting close to the end of the semester, what are one or two things that you are most proud of from this project?

The thing that I feel the most proud of is that within about half a semester we were able to come together as a team and make a pretty well functioning chrome extension which none of us have really done before.
